File: classTypes4/java/src/AnimalBase.java
public abstract class AnimalBase implements Animal {
    // atributos comunes
    private String nombreCientifico;
    protected double pesoKg; // Usamos protected para solo que las subclases puedan acceder

    // constructor para inicializar atributos
    public AnimalBase(String nombreCientifico, double pesoKg) {
        this.nombreCientifico = nombreCientifico;
        this.pesoKg = pesoKg;
    }

    // implementación de el método de la interfaz Animal
    @Override
    public void mover() {
        System.out.println("El animal se está moviendo con sus patas.");
    }
    
    // lo dejamos abstracto porque cada animal emite un sonido diferente
    @Override
    public abstract void emitirSonido();
    
    // Getter para nombreCientifico
    public String getNombreCientifico() {
        return nombreCientifico;
    }
}

File: classTypes4/c#/AnimalBase.cs
public abstract class AnimalBase : IAnimal // implementa la interfaz
{
    // atributos comunes
    private string nombreCientifico;
    // usamos protected para solo que las subclases puedan acceder
    public double pesoKg; 

    // constructor para inicializar atributos
    public AnimalBase(string nombreCientifico, double pesoKg){
        this.nombreCientifico = nombreCientifico;
        this.pesoKg = pesoKg;
    }

    // implementacion del metodo de la interfaz ianimal (no es abstracta ni virtual, es una implementacion base)
    public void mover(){
        Console.WriteLine("el animal se esta moviendo con sus patas");
    }
    
    // lo dejamos abstracto porque cada animal emite un sonido diferente
    // el metodo en c# debe ser declarado como 'abstract' y no lleva cuerpo
    public abstract void emitirsonido(); 

    // getter para nombrecientifico
    public string getNombreCientifico(){
        return nombreCientifico;
    }
}